errors.go

```diff
@@ -10,26 +10,32 @@ type Error struct {
 	Msg  string
 }
 
+// Error converts an Error to a string
```

Comment:
I still don't get why we do not end every sentence with a dot.

![image](https://user-images.githubusercontent.com/20455996/163341889-d6e7ac4b-22fa-416a-a79c-cfbf3d2fe988.png)

Reply:
Fixed.


```diff
commit 2688b1e0e88baa0616193efd239c92864acfcf2c
Author: Sergey Bronnikov <sergeyb@tarantool.org>
Date:   Thu Apr 14 16:05:39 2022 +0300

    Add dots to the end of sentences [TO SQUASH]

diff --git a/client_tools.go b/client_tools.go
index ed76ea2..de10a36 100644
--- a/client_tools.go
+++ b/client_tools.go
@@ -4,7 +4,7 @@ import (
 	"gopkg.in/vmihailenco/msgpack.v2"
 )
 
-// IntKey is utility type for passing integer key to Select*, Update* and Delete*
+// IntKey is utility type for passing integer key to Select*, Update* and Delete*.
 // It serializes to array with single integer element.
 type IntKey struct {
 	I int
@@ -16,7 +16,7 @@ func (k IntKey) EncodeMsgpack(enc *msgpack.Encoder) error {
 	return nil
 }
 
-// UintKey is utility type for passing unsigned integer key to Select*, Update* and Delete*
+// UintKey is utility type for passing unsigned integer key to Select*, Update* and Delete*.
 // It serializes to array with single integer element.
 type UintKey struct {
 	I uint
@@ -28,7 +28,7 @@ func (k UintKey) EncodeMsgpack(enc *msgpack.Encoder) error {
 	return nil
 }
 
-// UintKey is utility type for passing string key to Select*, Update* and Delete*
+// UintKey is utility type for passing string key to Select*, Update* and Delete*.
 // It serializes to array with single string element.
 type StringKey struct {
 	S string
@@ -40,8 +40,8 @@ func (k StringKey) EncodeMsgpack(enc *msgpack.Encoder) error {
 	return nil
 }
 
-// IntIntKey is utility type for passing two integer keys to Select*, Update* and Delete*
-// It serializes to array with two integer elements
+// IntIntKey is utility type for passing two integer keys to Select*, Update* and Delete*.
+// It serializes to array with two integer elements.
 type IntIntKey struct {
 	I1, I2 int
 }
@@ -53,7 +53,7 @@ func (k IntIntKey) EncodeMsgpack(enc *msgpack.Encoder) error {
 	return nil
 }
 
-// Op - is update operation
+// Op - is update operation.
 type Op struct {
 	Op    string
 	Field int
diff --git a/connection.go b/connection.go
index e8b87fb..a999c16 100644
--- a/connection.go
+++ b/connection.go
@@ -29,16 +29,16 @@ type ConnEventKind int
 type ConnLogKind int
 
 const (
-	// Connected signals that connection is established or reestablished
+	// Connected signals that connection is established or reestablished.
 	Connected ConnEventKind = iota + 1
-	// Disconnected signals that connection is broken
+	// Disconnected signals that connection is broken.
 	Disconnected
-	// ReconnectFailed signals that attempt to reconnect has failed
+	// ReconnectFailed signals that attempt to reconnect has failed.
 	ReconnectFailed
-	// Either reconnect attempts exhausted, or explicit Close is called
+	// Either reconnect attempts exhausted, or explicit Close is called.
 	Closed
 
-	// LogReconnectFailed is logged when reconnect attempt failed
+	// LogReconnectFailed is logged when reconnect attempt failed.
 	LogReconnectFailed ConnLogKind = iota + 1
 	// LogLastReconnectFailed is logged when last reconnect attempt failed,
 	// connection will be closed after that.
@@ -48,7 +48,7 @@ const (
 	LogUnexpectedResultId
 )
 
-// ConnEvent is sent throw Notify channel specified in Opts
+// ConnEvent is sent throw Notify channel specified in Opts.
 type ConnEvent struct {
 	Conn *Connection
 	Kind ConnEventKind
@@ -122,7 +122,7 @@ type Connection struct {
 	// Schema contains schema loaded on connection.
 	Schema    *Schema
 	requestId uint32
-	// Greeting contains first message sent by Tarantool
+	// Greeting contains first message sent by Tarantool.
 	Greeting *Greeting
 
 	shard      []connShard
@@ -136,7 +136,7 @@ type Connection struct {
 	lenbuf  [PacketLengthBytes]byte
 }
 
-var _ = Connector(&Connection{}) // check compatibility with connector interface
+var _ = Connector(&Connection{}) // Check compatibility with connector interface.
 
 type connShard struct {
 	rmut     sync.Mutex
@@ -150,7 +150,7 @@ type connShard struct {
 	_pad   [16]uint64
 }
 
-// Greeting is a message sent by Tarantool on connect
+// Greeting is a message sent by Tarantool on connect.
 type Greeting struct {
 	Version string
 	auth    string
@@ -159,7 +159,7 @@ type Greeting struct {
 // Opts is a way to configure Connection
 type Opts struct {
 	// Timeout is requests timeout.
-	// Also used to setup net.TCPConn.Set(Read|Write)Deadline
+	// Also used to setup net.TCPConn.Set(Read|Write)Deadline.
 	Timeout time.Duration
 	// Reconnect is a pause between reconnection attempts.
 	// If specified, then when Tarantool is not reachable or disconnected,
@@ -170,9 +170,9 @@ type Opts struct {
 	// MaxReconnects is a maximum reconnect attempts.
 	// After MaxReconnects attempts Connection becomes closed.
 	MaxReconnects uint
-	// User name for authorization
+	// User name for authorization.
 	User string
-	// Pass is password for authorization
+	// Pass is password for authorization.
 	Pass string
 	// RateLimit limits number of 'in-fly' request, i.e. already put into
 	// requests queue, but not yet answered by server or timeouted.
@@ -198,9 +198,10 @@ type Opts struct {
 	// Notify is a channel which receives notifications about Connection status
 	// changes.
 	Notify chan<- ConnEvent
-	// Handle is user specified value, that could be retrivied with Handle() method
+        // Handle is user specified value, that could be retrivied with
+        // Handle() method.
 	Handle interface{}
-	// Logger is user specified logger used for error messages
+	// Logger is user specified logger used for error messages.
 	Logger Logger
 }
 
@@ -268,10 +269,10 @@ func Connect(addr string, opts Opts) (conn *Connection, err error) {
 			return nil, err
 		} else if ok && (ter.Code == ErrNoSuchUser ||
 			ter.Code == ErrPasswordMismatch) {
-			/* reported auth errors immediately */
+			// Reported auth errors immediately.
 			return nil, err
 		} else {
-			// without SkipSchema it is useless
+			// Without SkipSchema it is useless.
 			go func(conn *Connection) {
 				conn.mutex.Lock()
 				defer conn.mutex.Unlock()
@@ -288,7 +289,7 @@ func Connect(addr string, opts Opts) (conn *Connection, err error) {
 		go conn.timeouts()
 	}
 
-	// TODO: reload schema after reconnect
+	// TODO: reload schema after reconnect.
 	if !conn.opts.SkipSchema {
 		if err = conn.loadSchema(); err != nil {
 			conn.mutex.Lock()
@@ -412,7 +413,7 @@ func (conn *Connection) dial() (err error) {
 		}
 	}
 
-	// Only if connected and authenticated
+	// Only if connected and authenticated.
 	conn.lockShards()
 	conn.c = connection
 	atomic.StoreUint32(&conn.state, connConnected)
diff --git a/errors.go b/errors.go
index 68034d1..a6895cc 100644
--- a/errors.go
+++ b/errors.go
@@ -4,13 +4,13 @@ import (
 	"fmt"
 )
 
-// Error is wrapper around error returned by Tarantool
+// Error is wrapper around error returned by Tarantool.
 type Error struct {
 	Code uint32
 	Msg  string
 }
 
-// Error converts an Error to a string
+// Error converts an Error to a string.
 func (tnterr Error) Error() string {
 	return fmt.Sprintf("%s (0x%x)", tnterr.Msg, tnterr.Code)
 }
@@ -22,7 +22,7 @@ type ClientError struct {
 	Msg  string
 }
 
-// Error converts a ClientError to a string
+// Error converts a ClientError to a string.
 func (clierr ClientError) Error() string {
 	return fmt.Sprintf("%s (0x%x)", clierr.Msg, clierr.Code)
 }
@@ -45,7 +45,7 @@ func (clierr ClientError) Temporary() bool {
 	}
 }
 
-// Tarantool client error codes
+// Tarantool client error codes.
 const (
 	ErrConnectionNotReady = 0x4000 + iota
 	ErrConnectionClosed   = 0x4000 + iota
@@ -54,7 +54,7 @@ const (
 	ErrRateLimited        = 0x4000 + iota
 )
 
-// Tarantool server error codes
+// Tarantool server error codes.
 const (
 	ErrUnknown                       = 0   // Unknown error
 	ErrIllegalParams                 = 1   // Illegal parameters, %s
diff --git a/request.go b/request.go
index a0eda92..ff8a7e8 100644
--- a/request.go
+++ b/request.go
@@ -7,7 +7,7 @@ import (
 	"gopkg.in/vmihailenco/msgpack.v2"
 )
 
-// Future is a handle for asynchronous request
+// Future is a handle for asynchronous request.
 type Future struct {
 	requestId   uint32
 	requestCode int32
@@ -51,7 +51,7 @@ func (req *Future) fillInsert(enc *msgpack.Encoder, spaceNo uint32, tuple interf
 
 // Select performs select to box space.
 //
-// It is equal to conn.SelectAsync(...).Get()
+// It is equal to conn.SelectAsync(...).Get().
 func (conn *Connection) Select(space, index interface{}, offset, limit, iterator uint32, key interface{}) (resp *Response, err error) {
 	return conn.SelectAsync(space, index, offset, limit, iterator, key).Get()
 }
@@ -120,7 +120,7 @@ func (conn *Connection) Eval(expr string, args interface{}) (resp *Response, err
 	return conn.EvalAsync(expr, args).Get()
 }
 
-// single used for conn.GetTyped for decode one tuple
+// single used for conn.GetTyped for decode one tuple.
 type single struct {
 	res   interface{}
 	found bool
@@ -405,7 +405,7 @@ func (fut *Future) wait() {
 	<-fut.ready
 }
 
-// Get waits for Future to be filled and returns Response and error
+// Get waits for Future to be filled and returns Response and error.
 //
 // Response will contain deserialized result in Data field.
 // It will be []interface{}, so if you want more performace, use GetTyped method.
@@ -442,7 +442,7 @@ func init() {
 	close(closedChan)
 }
 
-// WaitChan returns channel which becomes closed when response arrived or error occured
+// WaitChan returns channel which becomes closed when response arrived or error occured.
 func (fut *Future) WaitChan() <-chan struct{} {
 	if fut.ready == nil {
 		return closedChan
diff --git a/response.go b/response.go
index 09cf12e..379f23d 100644
--- a/response.go
+++ b/response.go
@@ -10,7 +10,7 @@ type Response struct {
 	RequestId uint32
 	Code      uint32
 	Error     string // error message
-	// Data contains deserialized data for untyped requests
+	// Data contains deserialized data for untyped requests.
 	Data []interface{}
 	buf  smallBuf
 }
@@ -139,7 +139,7 @@ func (resp *Response) decodeBodyTyped(res interface{}) (err error) {
 	return
 }
 
-// String implements Stringer interface
+// String implements Stringer interface.
 func (resp *Response) String() (str string) {
 	if resp.Code == OkCode {
 		return fmt.Sprintf("<%d OK %v>", resp.RequestId, resp.Data)
diff --git a/schema.go b/schema.go
index 5e07d0d..29f34e7 100644
--- a/schema.go
+++ b/schema.go
@@ -7,9 +7,9 @@ import (
 // Schema contains information about spaces and indexes.
 type Schema struct {
 	Version uint
-	// Spaces is map from space names to spaces
+	// Spaces is map from space names to spaces.
 	Spaces map[string]*Space
-	// SpacesById is map from space numbers to spaces
+	// SpacesById is map from space numbers to spaces.
 	SpacesById map[uint32]*Space
 }
 
@@ -24,9 +24,9 @@ type Space struct {
 	FieldsCount uint32
 	Fields      map[string]*Field
 	FieldsById  map[uint32]*Field
-	// Indexes is map from index names to indexes
+	// Indexes is map from index names to indexes.
 	Indexes map[string]*Index
-	// IndexesById is map from index numbers to indexes
+	// IndexesById is map from index numbers to indexes.
 	IndexesById map[uint32]*Index
 }
 
@@ -36,7 +36,7 @@ type Field struct {
 	Type string
 }
 
-// Index contains information about index
+// Index contains information about index.
 type Index struct {
 	Id     uint32
 	Name   string
@@ -65,7 +65,7 @@ func (conn *Connection) loadSchema() (err error) {
 	schema.SpacesById = make(map[uint32]*Space)
 	schema.Spaces = make(map[string]*Space)
 
-	// reload spaces
+	// Reload spaces.
 	resp, err = conn.Select(vspaceSpId, 0, 0, maxSchemas, IterAll, []interface{}{})
 	if err != nil {
 		return err
@@ -118,7 +118,7 @@ func (conn *Connection) loadSchema() (err error) {
 		schema.Spaces[space.Name] = space
 	}
 
-	// reload indexes
+	// Reload indexes.
 	resp, err = conn.Select(vindexSpId, 0, 0, maxSchemas, IterAll, []interface{}{})
 	if err != nil {
 		return err
@@ -136,7 +136,7 @@ func (conn *Connection) loadSchema() (err error) {
 			opts := row[4].(map[interface{}]interface{})
 			var ok bool
 			if index.Unique, ok = opts["unique"].(bool); !ok {
-				/* see bug https://github.com/tarantool/tarantool/issues/2060 */
+				/* See bug https://github.com/tarantool/tarantool/issues/2060 */
 				index.Unique = true
 			}
 		default:
```